### DISCUSSION TITLE
Duplicated Typst References

### DISCUSSION THREAD

**Discussion by 0ceangypsy** [quarto-dev/quarto-cli#5502] [Q&A] [answered]
Hello, 

I'm using the typst format with: 
**Typsy CLI Version** 0.3.0
**Quarto Version** 1.4.51

I have a document, where I want to use references. I use Bibtex to store my references and everytime I add some reference it is printed double. 

Is there a way to print references only once? 

Here is my yaml

`---
format:
  typst-report-typst: default
  
bibliography: References.bib
bibliographystyle: "chicago-author-date"

---`

Here is the citation reference 

`is a reliability coefficient and measure of internal consistency of tests and measures [@Cronbach_1951]`

Here is the output screenshot 
<img width="744" alt="Screenshot 2023-05-11 at 17 15 18" src="https://github.com/quarto-dev/quarto-cli/assets/12598838/7ac45760-79bc-423c-9f38-fae8afb0f9d5">

By the way if I enter references as I would normally do with markdown `::: {#refs}
:::` the behavior is the same. However if I write it in the online platform of Typst I don't have a bibliography duplication issue. Any help is greatly appreciated.

**Comment by mcanouil** [comment]
Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by 0ceangypsy** [comment]
Sure here it is 

````qmd
---
title: "Reproducible example" 
author: "0ceangypsy"
date: today
format:
  typst-report-typst: default
  
execute:
  echo: false
  warning: false
  freeze: true
  
bibliography: References.bib
bibliographystyle: "chicago-author-date"

---

## Cronbach's Alpha

Cronbach's alpha (Cronbach's $\alpha$), also known as rho-equivalent reliability ($\rho_T$) or coefficient alpha (coefficient $\alpha$), is a reliability coefficient and measure of internal consistency of tests and measures [@Cronbach_1951]

The end.

````

Here are the contents of the bib file: 

````bib
@article{Cronbach_1951,
	doi = {10.1007/bf02310555},
	year = 1951,
	month = {sep},
	publisher = {Springer Science and Business Media {LLC}},
	vol = {16},
	number = {3},
	pages = {297--334},
	author = {Lee J. Cronbach},
	title = {Coefficient alpha and the internal structure of tests},
	journal = {Psychometrika}
}



````

Here is the _extension.yml 

```` yml
title: Typst-report
author: 0ceangypsy
version: 1.0.0
quarto-required: ">=1.4.0"
contributes:
  formats:
    typst:
      template-partials:
        - typst-template.typ
        - typst-show.typ
        - biblio.typ
````

Please let me know if that suffices for reproduction.

**Reply by 0ceangypsy** [reply]
I also tried a vanilla typst version without the _extension.yml 

Here is the code 

````
---
title: "Untitled"
format: typst

bibliography: References.bib
bibliographystyle: "chicago-author-date"
---

Hello [@Cronbach_1951]
````

Please keep in mind that the References.bib contains the entry as in my previous post. 

The issue I pointed out remains the same. 

Also I tried the following: 
````
---
title: "Untitled"
format: typst

bibliography: References.bib
bibliographystyle: "chicago-author-date"
---

Hello 
``````

leads to the generation of a Bibliography section even though there are no citations in the text. My understanding is that in this case, no bibliography section entry should be shown on the output. Am I missing something?

**Reply by cderv** [reply, marked as answer]
If I add `keep-typ: true` in the YAML, you can see the `.typ` file. There will be at the bottom of the file in the case with no reference,

````
#set bibliography(style: "chicago-author-date")

#bibliography("References.bib")
````

It seems it cause typst to create a Bibliography section anyway. 

When a citation is added, it seems Pandoc will add some bibliography content

````
#label("refs")
#label("ref-Cronbach_1951")
Cronbach, Lee J. 1951. “Coefficient Alpha and the Internal Structure of Tests.” #emph[Psychometrika], no. 3 (September): 297–334. #link("https://doi.org/10.1007/bf02310555").

#set bibliography(style: "chicago-author-date")


#bibliography("References.bib")
````

It seems from https://github.com/jgm/pandoc/issues/8763#issuecomment-1500387547 that it requires some tweak when citation processing is done by Pandoc. (which happens when citeproc is used)

So try adding `citeproc: false` to your YAML header
````markdown 
---
title: "Untitled"
format: typst

bibliography: References.bib
bibliographystyle: "chicago-author-date"
citeproc: false
---

Hello [@Cronbach_1951]
````

it should output as expected 

We have indeed something to improve here in our default as it seems we are using `citeproc`
